### ISSUE TITLE
Pattern matching should adhere to `csharp_new_line_before_open_brace = none` (IDE0055)

### ISSUE THREAD

**Issue by ashmind** [dotnet/roslyn#52413]
**Version Used**: 
Visual Studio 16.9.3

**Steps to Reproduce**:
See attached solution: [Bugs.Roslyn.PatternMatchingBraces.zip](https://github.com/dotnet/roslyn/files/6255806/Bugs.Roslyn.PatternMatchingBraces.zip)

1. Create `.editorconfig` with 
```editorconfig
[*.cs]
csharp_new_line_before_open_brace = none
dotnet_diagnostic.IDE0055.severity = error
```
2. Write something like:
```csharp
public bool IsAbc(Test test) {
    return test is {
        Name: "Abc"
    };
}
```

**Expected Behavior**:
IDE0055 has no concerns about the code as it matches editorconfig.

**Actual Behavior**:
IDE0055 is reported at `test is {`, suggested quickfix converts code into
```csharp
public bool IsAbc(Test test) {
    return test is
    {
        Name: "Abc"
    };
}
```

**Comment by mrahhal** [comment]
I would like to work on this.

Pattern matching doesn't seem to fall into one of the already existing `NewLineOption` values ([documented here](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/formatting-rules#csharp_new_line_before_open_brace)) below, so looks like this will require an addition to these values?

https://github.com/dotnet/roslyn/blob/1f837cc1845ee6366695a64d9530e87f75326c8d/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpFormattingOptions2.Parsers.cs#L141-L155
